[net8.0] Update stable branding

rmarinho opened the pull request:
### Description of Change

Update to stable 8.0.0 branding

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/aspnetcore build 20231024.15

Microsoft.AspNetCore.Authentication.Facebook , Microsoft.AspNetCore.Authentication.Google , Microsoft.AspNetCore.Authentication.MicrosoftAccount , Microsoft.AspNetCore.Authorization , Microsoft.AspNetCore.Components , Microsoft.AspNetCore.Components.Analyzers , Microsoft.AspNetCore.Components.Forms , Microsoft.AspNetCore.Components.Web , Microsoft.AspNetCore.Components.WebView , Microsoft.AspNetCore.Metadata , Microsoft.JSInterop
 From Version 8.0.0-rtm.23523.8 -> To Version 8.0.0-rtm.23524.15

rmarinho committed:
Bump other stuff to stable branding

rmarinho committed:
Merge branch 'net8.0' into update-stable-brading

jonathanpeppers commented on eng/Versions.props:
```diff
@@ -3,53 +3,53 @@
+    <MicrosoftAspNetCoreComponentsWebPackageVersion>8.0.0</MicrosoftAspNetCoreComponentsWebPackageVersion>
+    <MicrosoftAspNetCoreComponentsWebViewPackageVersion>8.0.0</MicrosoftAspNetCoreComponentsWebViewPackageVersion>
+    <MicrosoftAspNetCoreMetadataPackageVersion>8.0.0</MicrosoftAspNetCoreMetadataPackageVersion>
+    <MicrosoftJSInteropPackageVersion>8.0.0</MicrosoftJSInteropPackageVersion>
```
This might not build, it needs this below:
```xml
<DotNetMonoManifestVersionBand>$(DotNetVersionBand)</DotNetMonoManifestVersionBand>
<DotNetEmscriptenManifestVersionBand>$(DotNetVersionBand)</DotNetEmscriptenManifestVersionBand>
```
I'll merge my PR into this one: https://github.com/dotnet/maui/pull/18381

jonathanpeppers committed:
Bump to xamarin/xamarin-android/release/8.0.1xx@b48acafb 34.0.39 (#18381)

Changes: https://github.com/xamarin/xamarin-android/compare/ad4af58b...b48acafb
Changes: https://github.com/dotnet/runtime/compare/11ad607e...59edaad4

This also updates to 8.0.0 stable branded dotnet/runtime and dotnet/emsdk.

`.Transport` was renamed from dotnet/emsdk, so I manually ran:

    > darc update-dependencies --id 199105
    Looking up build with BAR id 199105
    Updating 'Microsoft.Android.Sdk.Windows': '34.0.36' => '34.0.39' (from build '8.0.1xx-b48acafb42921772e76464917d3c0900a009d680-1' of 'https://github.com/xamarin/xamarin-android')
    Checking for coherency updates...
    Using 'Strict' coherency mode. If this fails, a second attempt utilizing 'Legacy' Coherency mode will be made.
    Updating 'Microsoft.Dotnet.Sdk.Internal': '8.0.100-rtm.23522.1' => '8.0.100-rtm.23518.33' to ensure coherency with Microsoft.Android.Sdk.Windows@34.0.39
    Updating 'Microsoft.NETCore.App.Ref': '8.0.0-rtm.23520.16' => '8.0.0' to ensure coherency with Microsoft.Android.Sdk.Windows@34.0.39
    Updating 'Microsoft.NET.Workload.Emscripten.Current.Manifest-8.0.100': '8.0.0-rtm.23511.3' => '8.0.0' to ensure coherency with Microsoft.NETCore.App.Ref@8.0.0-rtm.23520.16
    Local dependencies updated based on build with BAR id 199105 (8.0.1xx-b48acafb42921772e76464917d3c0900a009d680-1 from https://github.com/xamarin/xamarin-android@release/8.0.1xx)

Then I chose the higher version of dotnet/installer, which was already
present here.

jonathanpeppers committed:
s!

jonathanpeppers committed:
.NET 6 and .NET 7 servicing feeds

Got these from: https://github.com/xamarin/xamarin-android/commit/f63ac29fb3e48bfb5785e9fad03a64f68f7fc750

jonathanpeppers committed:
Updating 'Microsoft.Android.Sdk.Windows': '34.0.39' => '34.0.40'

jonathanpeppers committed:
Fix NRT warnings

jonathanpeppers commented:
There are different iOS errors now:
```
/Users/builder/azdo/_work/1/s/src/Core/src/Platform/iOS/ApplicationExtensions.cs(26,4): error CA1422: This call site is reachable on: 'ios' 13.0 and later, 'maccatalyst' 13.0 and later, 'tvos' 13.0 and later. 'UIApplication.RequestSceneSessionActivation(UISceneSession?, NSUserActivity?, UISceneActivationRequestOptions?, Action<NSError>?)' is obsoleted on: 'ios' 17.0 and later (Please use 'ActivateSceneSession' instead.), 'maccatalyst' 17.0 and later (Please use 'ActivateSceneSession' instead.), 'tvos' 17.0 and later (Please use 'ActivateSceneSession' instead.). (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1422) [/Users/builder/azdo/_work/1/s/src/Core/src/Core.csproj::TargetFramework=net8.0-ios]
/Users/builder/azdo/_work/1/s/src/Core/src/Platform/iOS/PageViewController.cs(34,4): error CA1422: This call site is reachable on: 'iOS' 11.0 and later, 'maccatalyst' 11.0 and later. 'UIViewController.TraitCollectionDidChange(UITraitCollection?)' is obsoleted on: 'ios' 17.0 and later (Use the 'UITraitChangeObservable' protocol instead.), 'maccatalyst' 17.0 and later (Use the 'UITraitChangeObservable' protocol instead.). (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1422) [/Users/builder/azdo/_work/1/s/src/Core/src/Core.csproj::TargetFramework=net8.0-ios]
/Users/builder/azdo/_work/1/s/src/Core/src/Platform/iOS/PageViewController.cs(34,4): error CA1422: This call site is reachable on: 'MacCatalyst' 13.1 and later. 'UIViewController.TraitCollectionDidChange(UITraitCollection?)' is obsoleted on: 'maccatalyst' 17.0 and later (Use the 'UITraitChangeObservable' protocol instead.). (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1422) [/Users/builder/azdo/_work/1/s/src/Core/src/Core.csproj::TargetFramework=net8.0-maccatalyst]
/Users/builder/azdo/_work/1/s/src/Core/src/Platform/iOS/ApplicationExtensions.cs(26,4): error CA1422: This call site is reachable on: 'ios' 13.0 and later, 'maccatalyst' 13.0 and later, 'tvos' 13.0 and later. 'UIApplication.RequestSceneSessionActivation(UISceneSession?, NSUserActivity?, UISceneActivationRequestOptions?, Action<NSError>?)' is obsoleted on: 'ios' 17.0 and later (Please use 'ActivateSceneSession' instead.), 'maccatalyst' 17.0 and later (Please use 'ActivateSceneSession' instead.), 'tvos' 17.0 and later (Please use 'ActivateSceneSession' instead.). (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1422) [/Users/builder/azdo/_work/1/s/src/Core/src/Core.csproj::TargetFramework=net8.0-maccatalyst]
    0 Warning(s)
    4 Error(s)
```

rmarinho committed:
[iOS] Use new API for UISceneSessionActivation

rmarinho committed:
[iOS] Disable CA1422 for TraitCollectionDidChange is obsolete on iOS17

rmarinho committed:
[iOS] Disable warning on TraitCollectionDidChange on old renderers

rmarinho committed:
[tests] Fix Assetion on iOS

rmarinho committed:
Still missing some TraitCollectionDidChange
